File: Cryptography/python/caesar_cipher/testcases/test_caesar.py
#!/usr/bin/python

from caesar_cipher import *

def test_enc_dec():
	print 'Plaintext: ',
	plaintext = raw_input()

	print 'Key: ',
	key = int(input())

	c = CaesarCipher(key)
	ciphertext = c.encrypt(plaintext)
	dec_plaintext = c.decrypt(ciphertext)

	print 'Ciphertext: {0}'.format(ciphertext)
	print 'Decrypted plaintext: {0}'.format(dec_plaintext)


if __name__ == "__main__":
	test_enc_dec()

File: Cryptography/haskell/caesar_cipher/testcases/test_caesar.hs
import CaesarCipher

main = do
	plaintext <- (getLine::IO String)
	key <- (readLn::IO Int)
	
	let ciphertext = (caesar_encrypt plaintext key)
	putStrLn ("Encrypted Ciphertext: " ++ ciphertext)

	let dec_plaintext = (caesar_decrypt ciphertext key)
	putStrLn ("Decrypted Plaintext: " ++ dec_plaintext)

